src/cli/Merge.cpp

```diff
@@ -15,105 +15,91 @@
-
-    parser.addHelpOption();
-    parser.process(arguments);
-
-    const QStringList args = parser.positionalArguments();
-    if (args.size() != 2) {
-        errorTextStream << parser.helpText().replace("[options]", "merge [options]");
-        return EXIT_FAILURE;
-    }
-
-    auto db1 = Utils::unlockDatabase(args.at(0),
-                                     !parser.isSet(Command::NoPasswordOption),
-                                     parser.value(Command::KeyFileOption),
-                                     parser.isSet(Command::QuietOption) ? Utils::DEVNULL : Utils::STDOUT,
-                                     Utils::STDERR);
-    if (!db1) {
-        return EXIT_FAILURE;
-    }
+    const QStringList args = parser->positionalArguments();
 
     QSharedPointer<Database> db2;
-    if (!parser.isSet("same-credentials")) {
+    if (!parser->isSet(Merge::SameCredentialsOption)) {
         db2 = Utils::unlockDatabase(args.at(1),
-                                    !parser.isSet(noPasswordFromOption),
-                                    parser.value(keyFileFromOption),
-                                    parser.isSet(Command::QuietOption) ? Utils::DEVNULL : Utils::STDOUT,
+                                    !parser->isSet(Merge::NoPasswordFromOption),
+                                    parser->value(Merge::KeyFileFromOption),
+                                    parser->isSet(Command::QuietOption) ? Utils::DEVNULL : Utils::STDOUT,
                                     Utils::STDERR);
         if (!db2) {
             return EXIT_FAILURE;
         }
     } else {
         db2 = QSharedPointer<Database>::create();
         QString errorMessage;
-        if (!db2->open(args.at(1), db1->key(), &errorMessage, false)) {
+        if (!db2->open(args.at(1), database->key(), &errorMessage, false)) {
             errorTextStream << QObject::tr("Error reading merge file:\n%1").arg(errorMessage);
             return EXIT_FAILURE;
         }
     }
 
-    Merger merger(db2.data(), db1.data());
-    bool databaseChanged = merger.merge();
+    Merger merger(db2.data(), database.data());
+    QStringList changeList = merger.merge();
 
-    if (databaseChanged) {
+    for (QString mergeChange : changeList) {
+        outputTextStream << "\t" << mergeChange << endl;
+    }
+
+    if (!changeList.isEmpty() && !parser->isSet(Merge::DryRunOption)) {
         QString errorMessage;
-        if (!db1->save(args.at(0), &errorMessage, true, false)) {
+        if (!database->save(args.at(0), &errorMessage, true, false)) {
             errorTextStream << QObject::tr("Unable to save database to file : %1").arg(errorMessage) << endl;
             return EXIT_FAILURE;
         }
-        if (!parser.isSet(Command::QuietOption)) {
-            outputTextStream << "Successfully merged the database files." << endl;
-        }
-    } else if (!parser.isSet(Command::QuietOption)) {
+        outputTextStream << "Successfully merged the database files." << endl;
```

Comment:
Recommend we explicitly state: ```Successfully merged [DB1] into [DB2]``` replacing with the database filenames. That makes it clear that [DB2] is the database that was modified.

Reply:
@droidmonkey done in ab15b1f. I also renamed the database path variables and noticed we were not translating (`QObject.tr`) all the messages, so I fixed that as well.